### PR TITLE
[C-1254] Fix debounced navigation

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Refactors debounced navigation solution with a custom push function.
Inspiration here: https://github.com/react-navigation/react-navigation/issues/3839#issuecomment-377159605
dispatch docs: https://reactnavigation.org/docs/navigation-prop/#dispatch

### Dragons
Instead of using the nearest navigator, we are defaulting to the AppStack Navigator, which should result in consistent behavior. Without this, when we were in a collapsible tab bar, we didn't have access to the previous route, so it wouldn't work.

### CODE REVIEW

File: packages/mobile/src/screens/app-screen/AppTabNavigationProvider.tsx [hunk 1]
```diff
@@ -32,7 +31,7 @@ export const AppTabNavigationProvider = (
   props: AppTabNavigationProviderProps
 ) => {
   const { children } = props
-  const [navigation, setNavigation] = useState<Nullable<AppTabNavigation>>(null)
+  const [navigation, setNavigation] = useState({} as AppTabNavigation)
```
Comment (sliptype) [inline packages/mobile/src/screens/app-screen/AppTabNavigationProvider.tsx:34]: Why was this needed?

**Review by sliptype — APPROVED** [review]
Very interesting, great find

File: packages/mobile/src/screens/app-screen/AppTabNavigationProvider.tsx (hunk 1 above)
Comment (dylanjeffers) [reply packages/mobile/src/screens/app-screen/AppTabNavigationProvider.tsx:34]: yeah it's pretty bad, but since we are using this as the main navigator now, there is a moment where it's null as it's setting up, which was causing runtime issues. Was thinking this is a bit easier to do here, than cascade the null down to the useNavigate, which was messing with it's returned type, causing useNavigation() to return Nullable<Navigator>, so casted early. lmk though, open to suggestions

Comment (sliptype) [reply packages/mobile/src/screens/app-screen/AppTabNavigationProvider.tsx:34]: Oof, yeah this is probably fine for know
